### ISSUE TITLE
Unable to process p2p attestation

### ISSUE THREAD

**Issue by stefa2k** [prysmaticlabs/prysm#3664]
https://pastebin.com/5UeG3xt8

I used an outdated "latest" taged docker image to run the beacon chain. This might be the cause.

/E:
Also happens on "latest" taged docker image for beacon chain. Slightly differen stack trace though:
```
[2019-09-30 09:39:31] ERROR sync: Failed to handle p2p pubsub error=attestation aggregation signature did not verify
could not verify indexed attestation
github.com/prysmaticlabs/prysm/beacon-chain/blockchain/forkchoice.(*Store).verifyAttestation
        beacon-chain/blockchain/forkchoice/process_attestation.go:240
github.com/prysmaticlabs/prysm/beacon-chain/blockchain/forkchoice.(*Store).OnAttestation
        beacon-chain/blockchain/forkchoice/process_attestation.go:108
github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).ReceiveAttestationNoPubsub
        beacon-chain/blockchain/receive_attestation.go:66
github.com/prysmaticlabs/prysm/beacon-chain/sync.(*RegularSync).beaconAttestationSubscriber
        beacon-chain/sync/subscriber_beacon_attestation.go:17
github.com/prysmaticlabs/prysm/beacon-chain/sync.(*RegularSync).subscribe.func1
        beacon-chain/sync/subscriber.go:135
runtime.goexit
        GOROOT/src/runtime/asm_amd64.s:1357
could not process block from fork choice service
github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).ReceiveAttestationNoPubsub
        beacon-chain/blockchain/receive_attestation.go:68
github.com/prysmaticlabs/prysm/beacon-chain/sync.(*RegularSync).beaconAttestationSubscriber
        beacon-chain/sync/subscriber_beacon_attestation.go:17
github.com/prysmaticlabs/prysm/beacon-chain/sync.(*RegularSync).subscribe.func1
        beacon-chain/sync/subscriber.go:135
runtime.goexit
        GOROOT/src/runtime/asm_amd64.s:1357 topic=/eth2/beacon_attestation/ssz
```

**Comment by prestonvanloon** [comment]
Thanks for the bug report @stefa2k. What's going on here is that your client is receiving an attestation over p2p which cannot be verified in your local state. 

There are two scenarios this might happen:
- You or the peer producing the attestation are on a forked chain
- There is a software bug within your client or peer's client with regards to validator indices and/or assignments. Someone signed an attestation out of turn by improper instruction by the client.

**Comment by prestonvanloon** [comment]
We believe the issue has been temporarily resolved. The nodes that we run in the network we using an experimental cache which may have led to some issues. 

We'll be monitoring this the next day before closing.

**Comment by terencechain** [comment]
This was an issue due to incompatibility of 2 different cache implementations. 

Background:
We have been using caches since early 0.2 spec, the current implementations are outdated but still get the job done. We have 3 caches (i.e. active indices, shuffled indices and active balances) that are used in run time. Last week, we implemented a  newshuffled indices cache to replace the old implementations. The new design itself has better semantics, and well tested within the constraints (i.e. new cache <-> new cache) Unfortunately this particular issue raises up when the new cache node sync with old cache node. 

Will further investigate and provide updates to this thread

**Comment by terencechain** [comment]
Lowering the priority as it's no longer blocking run time

**Comment by prestonvanloon** [comment]
Related #3699

**Comment by prestonvanloon** [comment]
This is happening again. Looks like it is not related to the BLS pubkey cache which was disabled by default in #3699. Investigating...

**Comment by rauljordan** [comment]
Resolved by #3704
